### PR TITLE
Give rewards for completing secondary objectives of Informant Assault missions

### LINKED ISSUE #553: Stranded expedition mission improvements
- make default VIP an operative instead of a scientist with a lab coat
- swap rumor reward with soldier reward for if you rescue the unconscious soldier

### PR DESCRIPTION
**Pull request by cannonfodder1**
Closes #553

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc [hunk 1]
```diff
@@ -114,7 +117,97 @@ static function CreateInformantAssault (out array<X2DataTemplate> Templates)
...
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+	RewardTemplate = X2RewardTemplate(TemplateManager.FindStrategyElementTemplate('Reward_Soldier'));
+	ResHQ = class'UIUtilities_Strategy'.static.GetResistanceHQ();
+
+	class'X2ActivityTemplate_Assault'.static.CreateMission(NewGameState, ActivityState);
+
+	foreach NewGameState.IterateByClassType(class'XComGameState_MissionSite', MissionState)
+	{
+		break;
+	}
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc:138]:
```suggestion
	foreach NewGameState.IterateByClassType(class'XComGameState_MissionSite', MissionState)
	{
		if (MissionState.ObjectID != ActivityState.PrimaryObjectRef.ObjectID)
		{
			MissionState = none;
			continue;
		}
		
		// Found the correct one
		break;
	}
```

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc [hunk 2]
```diff
@@ -114,7 +117,97 @@ static function CreateInformantAssault (out array<X2DataTemplate> Templates)
...
+static function InformantAssaultSetup (XComGameState NewGameState, XComGameState_Activity ActivityState)
+{
+	local XComGameState_MissionSite MissionState;
+	local X2StrategyElementTemplateManager TemplateManager;
+	local X2RewardTemplate RewardTemplate;
+	local XComGameState_Reward RewardState;
+	local XComGameState_HeadquartersResistance ResHQ;
+
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+	RewardTemplate = X2RewardTemplate(TemplateManager.FindStrategyElementTemplate('Reward_Soldier'));
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc:130]:
```suggestion
	SoldierRewardTemplate = X2RewardTemplate(TemplateManager.FindStrategyElementTemplate('Reward_Soldier'));
```

**Comment by cannonfodder1** [comment]
The default VIP is determined by the arrObjectiveSpawnInfo array in XComMissionDefs.ini. It's set to scientist for most VIP missions.

**Comment by Xymanek** [comment]
But I thought you wanted to "make default VIP an operative instead of a scientist with a lab coat"?

**Comment by cannonfodder1** [comment]
> But I thought you wanted to "make default VIP an operative instead of a scientist with a lab coat"?

Yeah I just didn't get around to changing that last time I worked on this PR, I didn't request a review because I'm not done yet

**Comment by Xymanek** [comment]
Please mark as draft when still working on a PR

**Comment by Xymanek** [comment]
Could we use OPTC patching instead of config minus/plus? To avoid issues like in the other case

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc [hunk 3]
```diff
@@ -114,7 +117,104 @@ static function CreateInformantAssault (out array<X2DataTemplate> Templates)
...
+			continue;
+		}
+		
+		// Found the correct one
+		break;
+	}
+
+	if (MissionState == none)
+	{
+		`REDSCREEN("COULD NOT FETCH MISSION FROM CREATION STATE");
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultActivities.uc:149]:
```suggestion
		`REDSCREEN("CI InformantAssaultSetup: Could not fetch mission from creation state");
```

1) All caps doesn't help
2) Knowing **where** the error is from does help, especially as the message is very ambiguous on its own

**Pull request merged**